### PR TITLE
Add viewer module, Reduce preemptive bundling

### PR DESCRIPTION
**Pull request by datori**
Most significant changes should be reflected in the updated readme.

Essentially, this is just an improvement upon the previous model of using rollup for UMD -> ESM transform.

Additionally, some directory refactoring (to make usage more familiar to consumers of Mozilla's `pdfjs-dist`), and some minor syntax changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,21 +10,65 @@ npm install @bundled-es-modules/pdfjs-dist
...
 ```
 
-## Overview of this mirror repo
+**Important**: Unlike in the original build, the `pdfjs.GlobalWorkerOptions.workerSrc` is mandatory.
+
+# Overview
+
+### Background
+
+PDFJS is a library by Mozilla, created to facillitate client-side rendering of PDF documents.
```
Comment (robrez) [inline README.md]: This new background section is informative: UMD vs ESM. Do you think that the original summary above might be sufficient?  I sort of feel like "if you're here, you know why"?

File: README.md [hunk 2]
```diff
@@ -10,21 +10,65 @@ npm install @bundled-es-modules/pdfjs-dist
...
+│   ├── pdf.js                    - display layer
+│   └── pdf.worker.js             - core layer
+└── web/
+    ├── pdf_viewer.js             - top-level viewer import, containing all necessary setup
+    ├── module.js                 - pdf-viewer module, transformed from UMD to ESM
+    └── util.js                   - utility module for setup
+
+```
+
+### Rollup configuration
```
Comment (robrez) [inline README.md]: My preference would be to leave this commentary inside the rollup config itself.  I like the idea of outlining intent, but most consumers probably won't care so long as the output works as intended.

File: package.json [hunk 3]
```diff
@@ -26,5 +26,9 @@
   },
   "scripts": {
     "build": "rollup -c ./rollup.config.js"
+  },
+  "dependencies": {
```
Comment (robrez) [inline package.json]: Move to devDependencies

File: package-lock.json [hunk 4]
```diff
@@ -1,1933 +0,0 @@
-{
```
Comment (robrez) [inline package-lock.json]: I don't think we should delete package-lock

File: rollup.config.js [hunk 5]
```diff
@@ -1,53 +1,75 @@
-import resolve from 'rollup-plugin-node-resolve';
```
Comment (robrez) [inline rollup.config.js]: it looks like all single quotes were changed to double quotes.  I don't have a strong preference here but other repos in this org seem to use single quotes in rollup config.

Comment (datori) [reply rollup.config.js]: Changed to single quotes to abide by project

File: README.md (hunk 1 above)
Comment (datori) [reply README.md]: Shortened the background section to simply state what PDFJS is, and describe the original repository we're mirroring.

**Comment by datori** [comment]
All changes made, commits squashed

**Comment by robrez** [comment]
Looks good to me... I'll merge this a bit later if noone raises any concern

**Pull request merged**